### PR TITLE
exercise two

### PR DESCRIPTION
**Pull request by maz4**

### CODE REVIEW

**Review by mibu5 — CHANGES_REQUESTED** [review]
I've run the code and noticed a couple of things missing:
1) The Form should be active as well
2) When you click on listings it should download 
3) It was not in the task, but, add functionality to close the card with a close.

**Comment by mibu5** [comment]
When you click on the pin it should stay active till you close the card panel
In Project requirements: 
```
Add .map__pin--active class to an active
pin.
```

File: js/map.js [hunk 1]
```diff
@@ -134,21 +138,65 @@
...
+    return;
+  }
+
+  function closeCardHandler(event) {
+    if ((event.target.tagName === 'BUTTON') && event.target.classList.contains('popup__close')) {
+      removeCard();
+    }
+  }
+
+  mapPinsMainElement.addEventListener('mouseup', mouseUpHendler);
```
Comment (mibu5) [inline js/map.js]: mouseUpHendler -> mouseUpHandler

**Comment by mibu5** [comment]
Remember that according to the requirements, form fields are also inactive in inactive state. In
the markup the fields are active, so we need to deactivate them, by adding disabled
attribute to the inputs themselves or to the parent fieldsets either through DOM or
manually.

File: js/map.js [hunk 2]
```diff
@@ -134,21 +138,65 @@
...
 
   function renderCard(data, elementContainer) {
     elementContainer.insertBefore(data, elementContainer.firstElementChild);
   }
 
-  mapElement.classList.remove('map--faded');
+  function mouseUpHendler() {
+    mapElement.classList.remove('map--faded');
+    adFormContainer.classList.remove('ad-form--disabled');
+    updateAdress(getPinLocation(mapElement, '.map__pin--main', PIN_HIGHT, HALF_PIN_WIDTH), formAddressInput);
```
Comment (mibu5) [inline js/map.js]: updateAdress -> updateAddress

Comment (mibu5) [reply js/map.js]: I would also suggest for readability don't pass function with arguments as arguments. Don't save on variables in that case.

Comment (mibu5) [reply js/map.js]: You don't need to pass formAddressInput, it is global

File: js/map.js [hunk 3]
```diff
@@ -134,21 +138,65 @@
...
+    renderPins(generatePins(pinElementTemplate, nearbyLisitings), mapPinsElement);
+  }
+
+  function updateAdress(position, element) {
+    element.value = position.top + ', ' + position.left;
+  }
+
+  function getPinLocation(container, pinName, pinHight, halfPinWidth) {
+    var pinContainerElement = container.querySelector(pinName);
```
Comment (mibu5) [inline js/map.js]: get the pin from mouseUpHandler event, event.currentTarget, it's the same element as mapPinsMainElement

File: js/map.js [hunk 4]
```diff
@@ -134,21 +138,65 @@
...
+    return {
+      top: (parseFloat(getComputedStyle(pinContainerElement).top) + parseFloat(getComputedStyle(pinContainerElement, 'after').top) + pinHight),
```
Comment (mibu5) [inline js/map.js]: Maybe we can use `offsetTop` and offsetLeft instead of getting getComputedStyle(pinContainerElement) ?

File: js/map.js [hunk 5]
```diff
@@ -134,21 +138,65 @@
...
+      left: (parseFloat(getComputedStyle(pinContainerElement).left) + parseFloat(getComputedStyle(pinContainerElement, 'after').left) + halfPinWidth)
```
Comment (mibu5) [inline js/map.js]: Also, use variables, you're overusing func1(funct(args..), func(args..), arg) pattern, it's hard to read

File: js/map.js [hunk 6]
```diff
@@ -134,21 +138,65 @@
...
+    };
+  }
+
+  function showCard(elem, arr) {
+    renderCard(generateCard(arr[+elem.dataset.index], cardElementTemplate), filterElementContainer);
```
Comment (mibu5) [inline js/map.js]: why are you adding cards to the filterElementContainer? I think it's a mistake. It's a small filter at the bottom of the map

File: js/map.js [hunk 7]
```diff
@@ -134,21 +138,65 @@
...
+      if (filterElementContainer.querySelector('.map__card')) {
+        removeCard();
+      }
+      showCard(button, nearbyLisitings);
+    }
+    return;
+  }
+
+  function closeCardHandler(event) {
+    if ((event.target.tagName === 'BUTTON') && event.target.classList.contains('popup__close')) {
```
Comment (mibu5) [inline js/map.js]: don't use deligation here, set on the card when you open it, and remove the listener when you close the card. You know how to do it from this weeks meeting.

File: js/map.js [hunk 8]
```diff
@@ -134,21 +138,65 @@
...
+  function showCard(elem, arr) {
+    renderCard(generateCard(arr[+elem.dataset.index], cardElementTemplate), filterElementContainer);
+  }
+
+  function removeCard() {
+    filterElementContainer.querySelector('.map__card').remove();
+  }
+
+  function pinsHandler(event) {
+    var button = event.target.parentElement;
```
Comment (mibu5) [inline js/map.js]: bug you code only works when you click on an image, if you click on the border it won't open the popup.

File: js/map.js [hunk 9]
```diff
@@ -134,21 +138,65 @@
...
+    if (button.tagName === 'BUTTON' && button.hasAttribute('data-index')) {
+      if (filterElementContainer.querySelector('.map__card')) {
```
Comment (mibu5) [inline js/map.js]: this check should belongs inside the remove function.

File: js/map.js (hunk 8 above)
Comment (mibu5) [reply js/map.js]: your code should be along this lines:
```
while (target != mapElement) {
    if (target.tagName == 'BUTTON') {
      removeCard();
      showCard(button, nearbyLisitings);
      return;
    }
target = target.parentNode;
}

or while (target != this) { ...} as this is the current target, the listener attached to.
    
```
This is an alternative to closest

**Review by mibu5 — CHANGES_REQUESTED** [review]

File: js/map.js (hunk 1 above)
Comment (maz4) [reply js/map.js]: done

File: js/map.js (hunk 2 above)
Comment (maz4) [reply js/map.js]: done

File: js/map.js (hunk 3 above)
Comment (maz4) [reply js/map.js]: done

File: js/map.js (hunk 8 above)
Comment (maz4) [reply js/map.js]: done

File: js/map.js (hunk 9 above)
Comment (maz4) [reply js/map.js]: done

File: js/map.js (hunk 4 above)
Comment (maz4) [reply js/map.js]: done

File: js/map.js (hunk 5 above)
Comment (maz4) [reply js/map.js]: done

File: js/map.js (hunk 6 above)
Comment (maz4) [reply js/map.js]: done

File: js/map.js (hunk 7 above)
Comment (maz4) [reply js/map.js]: done

**Review by mibu5 — APPROVED** [review]

**Pull request merged**